File: trunk/sources/net.sf.j2s.java.org.eclipse.swt/src/org/eclipse/swt/widgets/About.java
/*******************************************************************************
 * Copyright (c) 2007 java2script.org and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Zhou Renjian - initial API and implementation
 *******************************************************************************/

package org.eclipse.swt.widgets;

import org.eclipse.swt.SWT;
import org.eclipse.swt.events.DisposeEvent;
import org.eclipse.swt.events.DisposeListener;
import org.eclipse.swt.events.SelectionAdapter;
import org.eclipse.swt.events.SelectionEvent;
import org.eclipse.swt.graphics.Font;
import org.eclipse.swt.graphics.Image;
import org.eclipse.swt.graphics.Point;
import org.eclipse.swt.internal.ResizeSystem;
import org.eclipse.swt.layout.GridData;
import org.eclipse.swt.layout.GridLayout;
import org.eclipse.swt.widgets.Button;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.Control;
import org.eclipse.swt.widgets.Display;
import org.eclipse.swt.widgets.Label;
import org.eclipse.swt.widgets.Link;
import org.eclipse.swt.widgets.Shell;

/**
 * 
 * @author Zhou Renjian
 *
 * Aug 3, 2008
 */
public class About {

	/**
	 * Launch the application
	 * @param args
	 */
	public static void openAbout(Shell objShell) {
		try {
			Shell aboutShell = null;
			if (objShell == null) {
				aboutShell = new Shell(SWT.TOOL);
			} else {
				aboutShell = new Shell(objShell, SWT.TOOL);
			}
			aboutShell.setText("About Java2Script");

			createContents(aboutShell);
			aboutShell.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
			final GridLayout gridLayout = new GridLayout();
			gridLayout.verticalSpacing = 0;
			gridLayout.marginWidth = 0;
			gridLayout.marginHeight = 0;
			gridLayout.horizontalSpacing = 0;
			aboutShell.setLayout(gridLayout);

			aboutShell.setMinimumSize(442, 301);
			aboutShell.open();
			aboutShell.layout();
			aboutShell.pack();
			Point size = aboutShell.getSize();
			int y = (aboutShell.getMonitor().clientHeight - size.y) / 2 - 20;
			if (y < 0) {
				y = 0;
			}
			aboutShell.setLocation((aboutShell.getMonitor().clientWidth - size.x) / 2, y);
			ResizeSystem.register(aboutShell, SWT.CENTER);
			aboutShell.addDisposeListener(new DisposeListener() {
				public void widgetDisposed(DisposeEvent e) {
					ResizeSystem.unregister((Shell) e.widget, SWT.CENTER);
				}
			});
			Display disp = aboutShell.display;
			while (!aboutShell.isDisposed()) {
				if (!aboutShell.display.readAndDispatch())
					aboutShell.display.sleep();
			}
			if (objShell == null) {
				disp.dispose();
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	/**
	 * Create contents of the window
	 */
	private static void createContents(Shell aboutShell) {
		final Composite composite = new Composite(aboutShell, SWT.NONE);
		composite.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		composite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
		final GridLayout gridLayout = new GridLayout();
		gridLayout.numColumns = 2;
		composite.setLayout(gridLayout);

		final Label logo = new Label(composite, SWT.NONE);
		final GridData gd_logo = new GridData(SWT.LEFT, SWT.TOP, false, false);
		gd_logo.heightHint = 160;
		gd_logo.widthHint = 160;
		logo.setLayoutData(gd_logo);
		Image logoImage = null;
		/**
		 * @j2sNative
		 * logoImage = new $wt.graphics.Image ($wt.widgets.Display.getCurrent (), $wt.widgets.Display.getResourceAsStream ("images/j2s-logo.gif"));
		 */ {}
		//logo.setImage(SWTResourceManager.getImage(About.class, "images/j2s-logo.gif"));
		logo.setImage(logoImage);
		
		logo.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));

		final Composite description = new Composite(composite, SWT.NONE);
		description.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		description.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
		description.setLayout(new GridLayout());

		final Label java2scriptLabel = new Label(description, SWT.NONE);
		Font j2sFont = null;
		/**
	     * @j2sNative
	     * j2sFont = new $wt.graphics.Font ($wt.widgets.Display.getCurrent (),  new $wt.graphics.FontData ("Arial", 18, 1)); // 1 = SWT.BOLD
	     */ {}
		//java2scriptLabel.setFont(SWTResourceManager.getFont("Arial", 18, SWT.BOLD));
		java2scriptLabel.setFont(j2sFont);
		
		java2scriptLabel.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		GridData gd_Java2Script = new GridData(SWT.FILL, SWT.CENTER, true, false);
		gd_Java2Script.verticalIndent = 6;
		gd_Java2Script.heightHint = 24;
		java2scriptLabel.setLayoutData(gd_Java2Script);
		java2scriptLabel.setText("Java2Script");

		final Label java2scriptj2sPacemakerLabel = new Label(description, SWT.WRAP);
		j2sFont = null;
		/**
	     * @j2sNative
	     * j2sFont = new $wt.graphics.Font ($wt.widgets.Display.getCurrent (),  new $wt.graphics.FontData ("Arial", 10, 0));
	     */ {}
	    java2scriptj2sPacemakerLabel.setFont(j2sFont);
		java2scriptj2sPacemakerLabel.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		java2scriptj2sPacemakerLabel.setLayoutData(new GridData(225, SWT.DEFAULT));
		java2scriptj2sPacemakerLabel.setText("Java2Script (J2S) open source project provides an Eclipse Java to JavaScript compiler plugin, provides java.lang.*, java.util.* and other common utilities, provides an JavaScript implementation of Eclipse Standard Widget Toolkit (SWT), supports SWT-based Rich Client Platform (RCP) to Rich Internet Application (RIA) conversions.");

		final Link homepageLink = new Link(description, SWT.NONE);
		homepageLink.setFont(j2sFont);
		final GridData gd_homepageLink = new GridData();
		gd_homepageLink.verticalIndent = 8;
		homepageLink.setLayoutData(gd_homepageLink);
		homepageLink.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		homepageLink.setText("Home: <a href=\"http://java2script.org/\">java2script.org</a> / <a href=\"http://j2s.sourceforge.net/\">j2s.sourceforge.net</a>");

		final Link founderLink = new Link(description, SWT.NONE);
		founderLink.setFont(j2sFont);
		founderLink.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		founderLink.setText("Founder: <a href=\"http://zhourenjian.name/\">Zhou Renjian</a>");

		Composite blank = new Composite(description, SWT.NONE);
		blank.setBackground(Display.getCurrent().getSystemColor(SWT.COLOR_WHITE));
		blank.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
		blank.setLayout(new GridLayout());

		final Button okButton = new Button(description, SWT.NONE);
		final GridData gd_okButton = new GridData(SWT.RIGHT, SWT.CENTER, false, false);
		gd_okButton.widthHint = 120;
		okButton.setLayoutData(gd_okButton);
		okButton.setText("OK");
		okButton.addSelectionListener(new SelectionAdapter() {
		
			public void widgetSelected(SelectionEvent e) {
				((Control) e.widget).getShell().close();
			}
		
		});
		//
	}

}

File: trunk/incubator/net.sf.j2s.test.mauve/j2slib/org/eclipse/swt/widgets/About.js
$_L(null,"$wt.widgets.About",["$wt.events.DisposeListener","$.SelectionAdapter","$wt.internal.ResizeSystem","$wt.layout.GridData","$.GridLayout","$wt.widgets.Button","$.Composite","$.Display","$.Label","$.Link","$.Shell"],function(){
c$=$_T($wt.widgets,"About");
c$.openAbout=$_M(c$,"openAbout",
function(objShell){
try{
var aboutShell=null;
if(objShell==null){
aboutShell=new $wt.widgets.Shell(4);
}else{
aboutShell=new $wt.widgets.Shell(objShell,4);
}aboutShell.setText("About Java2Script");
$wt.widgets.About.createContents(aboutShell);
aboutShell.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
var gridLayout=new $wt.layout.GridLayout();
gridLayout.verticalSpacing=0;
gridLayout.marginWidth=0;
gridLayout.marginHeight=0;
gridLayout.horizontalSpacing=0;
aboutShell.setLayout(gridLayout);
aboutShell.setMinimumSize(442,301);
aboutShell.open();
aboutShell.layout();
aboutShell.pack();
var size=aboutShell.getSize();
var y=Math.floor((aboutShell.getMonitor().clientHeight-size.y)/2)-20;
if(y<0){
y=0;
}aboutShell.setLocation(Math.floor((aboutShell.getMonitor().clientWidth-size.x)/2),y);
$wt.internal.ResizeSystem.register(aboutShell,16777216);
aboutShell.addDisposeListener((($_D("$wt.widgets.About$1")?0:org.eclipse.swt.widgets.About.$About$1$()),$_N($wt.widgets.About$1,this,null)));
var disp=aboutShell.display;
Sync2Async.block(aboutShell,this,function(){
if(objShell==null){
disp.dispose();
}});
return;
}catch(e){
if($_O(e,Exception)){
e.printStackTrace();
}else{
throw e;
}
}
},"$wt.widgets.Shell");
c$.createContents=$_M(c$,"createContents",
($fz=function(aboutShell){
var composite=new $wt.widgets.Composite(aboutShell,0);
composite.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
composite.setLayoutData(new $wt.layout.GridData(4,4,true,true));
var gridLayout=new $wt.layout.GridLayout();
gridLayout.numColumns=2;
composite.setLayout(gridLayout);
var logo=new $wt.widgets.Label(composite,0);
var gd_logo=new $wt.layout.GridData(16384,128,false,false);
gd_logo.heightHint=160;
gd_logo.widthHint=160;
logo.setLayoutData(gd_logo);
var logoImage=null;
{
logoImage=new $wt.graphics.Image($wt.widgets.Display.getCurrent(),$wt.widgets.Display.getResourceAsStream("images/j2s-logo.gif"));
}logo.setImage(logoImage);
logo.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
var description=new $wt.widgets.Composite(composite,0);
description.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
description.setLayoutData(new $wt.layout.GridData(4,4,true,true));
description.setLayout(new $wt.layout.GridLayout());
var java2scriptLabel=new $wt.widgets.Label(description,0);
var j2sFont=null;
{
j2sFont=new $wt.graphics.Font($wt.widgets.Display.getCurrent(),new $wt.graphics.FontData("Arial",18,1));
}java2scriptLabel.setFont(j2sFont);
java2scriptLabel.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
var gd_Java2Script=new $wt.layout.GridData(4,16777216,true,false);
gd_Java2Script.verticalIndent=6;
gd_Java2Script.heightHint=24;
java2scriptLabel.setLayoutData(gd_Java2Script);
java2scriptLabel.setText("Java2Script");
var java2scriptj2sPacemakerLabel=new $wt.widgets.Label(description,64);
j2sFont=null;
{
j2sFont=new $wt.graphics.Font($wt.widgets.Display.getCurrent(),new $wt.graphics.FontData("Arial",10,0));
}java2scriptj2sPacemakerLabel.setFont(j2sFont);
java2scriptj2sPacemakerLabel.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
java2scriptj2sPacemakerLabel.setLayoutData(new $wt.layout.GridData(225,-1));
java2scriptj2sPacemakerLabel.setText("Java2Script (J2S) open source project provides an Eclipse Java to JavaScript compiler plugin, provides java.lang.*, java.util.* and other common utilities, provides an JavaScript implementation of Eclipse Standard Widget Toolkit (SWT), supports SWT-based Rich Client Platform (RCP) to Rich Internet Application (RIA) conversions.");
var homepageLink=new $wt.widgets.Link(description,0);
homepageLink.setFont(j2sFont);
var gd_homepageLink=new $wt.layout.GridData();
gd_homepageLink.verticalIndent=8;
homepageLink.setLayoutData(gd_homepageLink);
homepageLink.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
homepageLink.setText("Home: <a href=\"http://java2script.org/\">java2script.org</a> / <a href=\"http://j2s.sourceforge.net/\">j2s.sourceforge.net</a>");
var founderLink=new $wt.widgets.Link(description,0);
founderLink.setFont(j2sFont);
founderLink.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
founderLink.setText("Founder: <a href=\"http://zhourenjian.name/\">Zhou Renjian</a>");
var blank=new $wt.widgets.Composite(description,0);
blank.setBackground($wt.widgets.Display.getCurrent().getSystemColor(1));
blank.setLayoutData(new $wt.layout.GridData(4,4,true,true));
blank.setLayout(new $wt.layout.GridLayout());
var okButton=new $wt.widgets.Button(description,0);
var gd_okButton=new $wt.layout.GridData(131072,16777216,false,false);
gd_okButton.widthHint=120;
okButton.setLayoutData(gd_okButton);
okButton.setText("OK");
okButton.addSelectionListener((($_D("$wt.widgets.About$2")?0:org.eclipse.swt.widgets.About.$About$2$()),$_N($wt.widgets.About$2,this,null)));
},$fz.isPrivate=true,$fz),"$wt.widgets.Shell");
c$.$About$1$=function(){
$_H();
c$=$_W($wt.widgets,"About$1",null,$wt.events.DisposeListener);
$_V(c$,"widgetDisposed",
function(e){
$wt.internal.ResizeSystem.unregister(e.widget,16777216);
},"$wt.events.DisposeEvent");
c$=$_P();
};
c$.$About$2$=function(){
$_H();
c$=$_W($wt.widgets,"About$2",$wt.events.SelectionAdapter);
$_V(c$,"widgetSelected",
function(e){
(e.widget).getShell().close();
},"$wt.events.SelectionEvent");
c$=$_P();
};
});
